components/common/autoTeleport/AutoTeleportModal.vue

```diff
@@ -0,0 +1,263 @@
+  const interaction =
+    props.transactions.actions[0].interaction?.toLocaleLowerCase()
+  return $i18n.t(`autoTeleport.steps.${interaction}.header`)
+})
+
+const steps = computed<TransactionStep[]>(() => {
+  return [
+    {
+      title: $i18n.t('autoTeleport.steps.1.title'),
+      subtitle: $i18n.t('autoTeleport.steps.1.subtitle'),
+      status: props.transactions.teleport.status.value,
+      isError: props.transactions.teleport.isError.value,
+      txId: props.transactions.teleport.txId.value,
+      prefix: props.transition.source?.prefix,
+      withAction: true,
+      retry: () => emit('telport:retry'),
+    },
+    {
+      title: $i18n.t('autoTeleport.steps.2.title'),
+      subtitle: $i18n.t('autoTeleport.steps.2.subtitle'),
+      stepStatus: checkBalanceState.value,
+    },
+    props.transactions.actions.map((action) => {
+      const { title, subtitle } = getActionDetails(action.interaction)
+      return {
+        title,
+        subtitle,
+        status: action.status.value,
+        isError: action.isError.value,
+        txId: action.txId.value,
+        blockNumber: action.blockNumber?.value,
+        isLoading: action.isLoading.value,
+        prefix: props.transition.destination?.prefix,
+        withAction: true,
+        retry: () => emit('action:retry', action.interaction),
+      }
+    }),
+  ].flat()
+})
+
+const handleActiveStep = (step) => {
+  activeStep.value = step
+}
+
+const btnDisabled = computed(() => {
+  return !actionsFinalized.value
+})
+
+const actionsFinalized = computed(() =>
+  props.transactions.actions
+    .map((action) => action.status)
+    .every((status) => status.value === TransactionStatus.Finalized),
+)
+
+const btnLabel = computed(() => {
+  if (!props.canDoAction || !activeStepInteraction.value) {
+    return $i18n.t('autoTeleport.finishAllStepsFirst')
+  }
+
+  if (!actionsFinalized.value) {
+    return getActionDetails(activeStepInteraction.value).submit
+  }
+
+  return $i18n.t('autoTeleport.close')
+})
+
+const getActionDetails = (interaction: string) => {
+  const i = interaction.toLocaleLowerCase()
+  return {
+    title: $i18n.t(`autoTeleport.steps.${i}.title`),
+    subtitle: $i18n.t(`autoTeleport.steps.${i}.subtitle`),
+    submit: $i18n.t(`autoTeleport.steps.${i}.submit`),
+  }
+}
+
+const FIRST_ACTION_STEP = 2
+const activeStepInteraction = computed(
+  () =>
+    props.transactions.actions[activeStep.value - FIRST_ACTION_STEP]
+      ?.interaction,
+)
+
+watch(activeStep, () => {
+  const isActionStep = activeStep.value >= FIRST_ACTION_STEP
+  if (isActionStep) {
+    emit('action:start', activeStepInteraction.value)
+  }
+})
+
+const submit = () => {
+  if (actionsFinalized.value) {
+    onClose()
+  }
+}
+
+const onClose = () => {
+  emit('close')
+}
+
+watch(actionsFinalized, () => {
+  if (actionsFinalized.value) {
+    emit('completed')
+
+    if (props.autoClose) {
+      setTimeout(() => {
+        onClose()
+      }, 3000)
+    }
```

Comment:
why use setTimeout

Reply:
for nft creation there is a redirect to the nft page, closing immedialty didn't feel quite right so that's why I added the auto close with a delay

Reply:
if so then move it to some const with a good name please
avoid magic numbers

Reply:
done